pkg/rancher/migrations.go

```diff
@@ -540,3 +543,69 @@ func insertOrUpdateCondition(d data.Object, desiredCondition summary.Condition)
 
 	return true, nil
 }
+
+func migrateCAPIKubeconfigs(w *wrangler.Context) error {
+	logrus.Info("Running CAPI secret migration")
+
+	mgmtClusters, err := w.Mgmt.Cluster().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, mgmtCluster := range mgmtClusters.Items {
+		logrus.Tracef("Checking if migration for cluster %s is needed", mgmtCluster.Name)
+
+		provClusters, err := w.Provisioning.Cluster().List(mgmtCluster.Spec.FleetWorkspaceName, metav1.ListOptions{})
```

Comment:
There is actually not a guaranteed 1:1 relationship between the namespace that the `provisioning` cluster lives in vs. the namespace of `mgmtCluster.Spec.FleetWorkspaceName`. You can specifically see the PR here: https://github.com/rancher/rancher/pull/42955 that added functionality that allows a user to change the fleet workspace name of a v2prov cluster without causing recreation of the resources from `fleet-default` (which is a misnomer, but a requirement that we were given and had to satisfy regardless of the limitations/bad UX).

I would rather say iterating over the provisioning clusters is a more "complete" solution in this case.

Reply:
Thanks for the feedback @Oats87 👍  This migration was changed to follow the pattern used by the **migrateMachinePoolsDynamicSchemaLabel** migration.

Are you suggesting that we go back to iterating over the namespaces and in turn iterating over the v1.Clusters in the namespaces?

Reply:
Giving this some further thought, do we even really need this migration? The kubeconfig secrets will all be eventually consistent as the controller will run through all of the kubeconfigs/clusters anyway on Rancher startup/when the controller manager becomes the leader.

Besides, from what I can tell, cluster-api will just error until the kubeconfig is properly labeled, and the version of cluster-api that requires the kubeconfig secrets to be labeled will not be installed until after the version of Rancher with this change starts anyway.

Reply:
@jakefhyde thoughts here?

Reply:
This is a good point, based on my understanding of the code as written, if the secret exists and doesn't have the label it would get deleted and recreated anyway.Is the goal of a migration here to prevent unintended side effects from deleting the secret since all cluster would need their secrets recreated?

Reply:
We added a migration to ensure the "at rest" secrets were migrated. See this: https://github.com/rancher/rancher/pull/42318#pullrequestreview-1565677338

Reply:
I've changed the logic to iterate around all the provisioning.v1.Clusters

Reply:
But the kubeconfig manager/corresponding controllers should migrate the kubeconfig secrets at startup as it'll consider the kubeconfig secret invalid on retrieval, which is why I'm thinking the migration is unnecessary.

Regardless, if you feel strongly we should have the migration in place, that's fine. We should definitely earmark it for removal in a future version of Rancher, once we properly implement version gates

Reply:
@Oats87 - i just tried what suggested. I removed the migration, leaving the check for valid kubeconfig and then deployed the new version of rancher. Within ~2mins the kubeconfigs for the existing clusters had been re-created with the new label.

I will update the 3 PRs to remove the migration.